src/middleware.js

```diff
@@ -2,11 +2,37 @@ import path from 'path';
 
 import mime from 'mime-types';
 
+import getETag from 'etag';
+
 import getFilenameFromUrl from './utils/getFilenameFromUrl';
 import handleRangeHeaders from './utils/handleRangeHeaders';
 import ready from './utils/ready';
 
+const { hasOwnProperty } = Object.prototype;
+
 export default function wrapper(context) {
+  const { compiler } = context;
+  const etagRegistry = new Map();
+
+  function computeEtags(stats) {
+    try {
+      const { assets } = stats.compilation;
+      for (const assetId in assets) {
+        if (hasOwnProperty.call(assets, assetId)) {
+          const { existsAt: fsPath } = assets[assetId];
+          const etag = getETag(assets[assetId].source());
```

Comment:
`assets[assetId]` is key for `WeakMap`

Reply:
Hmm, I'll need to be able to "fetch" this etag value whenever a new request comes in, and if we use the Asset instance as a key, I'm not sure if I'll be able to query the etag.

Would clearing the Map work?

Reply:
Sorry, no, clearing map is not fix memory leaking